### PR TITLE
Matlab example

### LINKED ISSUE #64: Create example in MATLAB
Using the MATLAB language, create an example

### PR DESCRIPTION
**Pull request by eranschweitzer**
### Summary

If merged this pull request will provide a demonstration of how to use Helics with Matlab via [matHELICS](https://github.com/GMLC-TDC/matHELICS)

This closes #64.

### Proposed changes

This is simply a duplication of the advanced default example to matlab.

### CODE REVIEW

**Comment by trevorhardy** [comment]
@eranschweitzer, does this produce identical results as the Python examples? I'm assuming not because of differences in implementation of the RNG.

**Comment by eranschweitzer** [comment]
Correct. The results are _similar_ but not _identical_. There might be a way to figure out how to get them identical because I believe the underlying random number generation algorithm is the same, but I don't think it's worth it.

Just for reference, here is the results for the Battery SOC in the python example: 
advanced_default_battery_SOCs_python


And here's the result in the matlab example:
advanced_default_battery_SOCs_matlab

File: user_guide_examples/advanced/advanced_default_matlab/Charger.m [hunk 1]
```diff
@@ -0,0 +1,201 @@
+
```
Comment (trevorhardy) [inline user_guide_examples/advanced/advanced_default_matlab/Charger.m:1]: Update comments to reflect @eranschweitzer wrote the MATLAB version

File: user_guide_examples/advanced/advanced_default_matlab/Controller.m [hunk 2]
```diff
@@ -0,0 +1,134 @@
+%{
```
Comment (trevorhardy) [inline user_guide_examples/advanced/advanced_default_matlab/Controller.m:1]: Add comments to reflect @eranschweitzer wrote the review

**Review by trevorhardy — APPROVED** [review]

**Comment by nightlark** [comment]
I’ve messed around some with creating a challenge related to predicting the
next “random” number in a sequence — what I found is that with the same
starting seed, the default random number generating algorithms for C/C++,
Go, and (iirc) Python produced different results.

**Pull request merged**